File: samples/sum_of_odds/sum_of_odds.py
#!/usr/bin/env python

'''
sum_of_odds.py - calculate sum of odd integers
'''

import sys

__version__ = 1.1
__date__ = '2021-01-13'

up_to = 170

def is_odd(x: int) -> bool:
  remainder = x % 2
  if remainder == 0:
    return False
  else:
    return True

def main():
  sum = number_of_odds = last_odd = 0

  for i in range(up_to + 1):
    if is_odd(i):
      sum += i
      number_of_odds += 1
      last_odd = i

  if sum == number_of_odds ** 2:
    if number_of_odds == 0:
      pass
    elif number_of_odds < 7:
      print(sum)
    else:
      print('1 + 3 + 5 + ... +', last_odd - 1,
          '+', last_odd, '=', sum)
  else:
    sys.stderr.write('Unexpected error: ' +
        str(sum) + "\n")

if __name__ == '__main__':
  main()

File: samples/sum_of_odds/sum_of_odds.r
#!/usr/bin/env Rscript

# sum_of_odds.r - calculate sum of odd integers

version__ = 1.1
date__ = '2021-01-14'

up_to <- 170

is_odd <- function(x)
{
  remainder <- x %% 2
  if (remainder == 0) return(FALSE)
  else                return(TRUE)
}

sum <- 0
number_of_odds <- 0
last_odd <- 0

for (i in 0:up_to)
{
  if (is_odd(i))
  {
    sum <- sum + i
    number_of_odds <- number_of_odds + 1
    last_odd <- i
  }
}

if (sum == number_of_odds ^ 2)
{
  if (number_of_odds == 0) { ; }
  else if (number_of_odds < 7) print(sum)
  else cat('1 + 3 + 5 + ... +', last_odd - 1,
           '+', last_odd, '=', sum, "\n")
} else warning('Unexpected error: ', sum)
